torch/csrc/jit/script/compiler.cpp

```diff
@@ -1298,246 +1302,92 @@ struct to_ir {
     }
   }
 
-  void emitForRange(
-      const SourceRange& range,
-      const Ident& target,
-      const List<Expr>& args,
-      const List<Stmt>& body) {
-    Value *end_val = nullptr, *start_val = nullptr, *step_val = nullptr;
-    bool isSimpleRange = (args.size() == 1);
-    std::vector<Value*> argVals;
-    for (auto i : args) {
-      argVals.push_back(ensureInt(range, emitExpr(i)));
-    }
-    if (isSimpleRange) {
-      end_val = argVals[0];
-      start_val = end_val->owningGraph()->insertConstant(0);
-      step_val = end_val->owningGraph()->insertConstant(1);
-      start_val->node()->setSourceRange(range);
-      end_val->node()->setSourceRange(range);
-    } else if (args.size() == 2 || args.size() == 3) {
-      start_val = argVals[0];
-      end_val = argVals[1];
-      if (args.size() == 3) {
-        step_val = argVals[2];
-      } else {
-        step_val = end_val->owningGraph()->insertConstant(1);
-        step_val->node()->setSourceRange(range);
+  // recursively assign the targets base on the iterable tree structure
```

Comment:
I do not think it is a good idea to special case the iterable_tree here. There are many places in the compiler that already do tuple assignment, e.g.
`a, (b, c)` = foo()`

We should be reusing that code for assign the loop lhs to whatever the result of getitem is.  The IterableTree can simply return a tree from that.

Reply:
does that mean we should make the `getitem` to return `SugaredValue` instead of `Value*`?